demos/apps/sleep/sleep.yaml

```diff
@@ -0,0 +1,32 @@
+apiVersion: v1
```

Comment:
The images are different. `docker.io/istio/app` is istio-specific image that originally created for integration tests. gcr.io/google_containers/echoserver:1.4 is kubernetes specific image (see https://github.com/kubernetes/contrib/tree/master/ingress/echoheaders). Both have their places. I used the general purpose image in the integrating-apps-with-istio task to make it clear no istio-specific changes are required to the app. 

re: source specific, I'm probably just misremembering the details, but I thought we didn't use source service because (a) pods may not necessarily have a service and (b) its a bit racy since a deployment could be created prior to the service.